fedmsg_meta_fedora_infrastructure/compose.py

```diff
@@ -27,88 +27,125 @@ class ComposeProcessor(BaseProcessor):
     __docs__ = "https://pagure.io/docs/releng"
     __obj__ = "Composes"
 
+    # dicts mapping topic substrings to subtitle event texts
+    # These are the old 'make-updates' topics
+    makeupevents = {
+        '.make-updates.':       "",
+        '.cloudimg-build.':     " building images for",
+        '.mash-atomic':         " Atomic updates mash for",
+        '.atomic-lorax':        " Atomic lorax run for",
+        '.cloudimg-checksum.':  " image checksum generation for",
+        '.cloudimg-staging.':   " master mirror publication for",
+    }
+    # as these events only existed prior to pungi 4 and the creation
+    # of different 'dists', we can always treat the shortname / dist
+    # as 'Fedora' for these
+    oldevents = {
+        '.mash.':               " mashing",
+        '.pungify.':            " building boot.iso for",
+        '.image.':              " building other images for",
+    }
+    oldevents.update(makeupevents)
+    # These ones existed after pungi 4, so we cannot make inferences
+    # about the dist / shortname; bare 'start' and 'complete' are
+    # implicitly in this set too
+    events = {
+        '.rsync.':              " master mirror publication for",
+    }
+    allevents = dict()
+    allevents.update(oldevents)
+    allevents.update(events)
+
     def subtitle(self, msg, **config):
 
+        # decide what event we're dealing with
+        gotevent = False
+        for event in self.allevents:
+            if event in msg['topic']:
```

Comment:
The old structure seemed to rely on ``.endswith()`` most of the time, that makes the topic a little more specific (cf my concern above)

Reply:
I didn't like the `endswith` because it meant there had to be two of everything, one for 'start' and one for 'complete'. This way seems cleaner, for me.